tests/emukit/core/test_outer_loop.py

```diff
@@ -43,6 +44,16 @@ def test_outer_loop(mock_next_point_calculator, mock_updater, mock_user_function
     assert(loop.loop_state.iteration == 2)
     assert(np.array_equal(loop.loop_state.X, np.array([[0], [0]])))
 
+
+def test_outer_loop_model_update(mock_next_point_calculator, mock_updater, mock_user_function):
+    """ Checks the model has the correct number of data points """
+
+    loop = OuterLoop(mock_next_point_calculator, mock_updater)
+    loop.run_loop(mock_user_function, 2)
+
+    # Check update was last called with a loop state with all the collected data points
+    assert mock_updater.update.call_args[0][0].X.shape[0] == 2
```

Comment:
Can you change this test so that it just checks that model has all the data instead? Tests that verify particular function calls are best avoided, as they test implementation details and not the output, and thus tend to be brittle.

Reply:
Hmmm, I'm not convinced. Then I'd need to instantiate a real model and a model updater instead of mocks. Would I create a dummy model class to do this? Actual models would slow the test down with unnecessary model optimizing. I'd still be testing as many implementation details as I am here, as I'd have to extract the data from the loop state.

Reply:
Have a mock model and implement update_data to just collect data. Then you can compare it with expected in the end, or just the length of it

Reply:
I've changed the test, could you take a look and see if that's what you meant? I feel like I'm now just testing the implementation of my mocks rather than whether the model updater gets the correct number of data points.

Reply:
In fact it is, thanks!